lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2438,6 +2493,50 @@ bool CppEmitter::hasBlockLabel(Block &block) {
   return blockMapper.count(&block);
 }
 
+template <typename ElTy>
+static bool hasSameDenseValue(DenseIntElementsAttr dense,
```

Comment:
My suggestion is that the caller will call dens.isSplat() and create a separate API for retrieving the first value as a string.